homeassistant/components/ourgroceries/todo.py

```diff
@@ -0,0 +1,117 @@
+"""A todo platform for OurGroceries."""
+
+import asyncio
+
+from homeassistant.components.todo import (
+    TodoItem,
+    TodoItemStatus,
+    TodoListEntity,
+    TodoListEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import OurGroceriesDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the OurGroceries todo platform config entry."""
+    coordinator: OurGroceriesDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        OurGroceriesTodoListEntity(coordinator, sl["id"], sl["name"])
+        for sl in coordinator.lists
+    )
+
+
+class OurGroceriesTodoListEntity(
```

Comment:
New integrations have to use `_attr_has_entity_name = True`. There are some docs on it but I'm on mobile right now.

Reply:
I thought that's only relevant when there's a device. Looking at other Todo implementations, they don't seem to be implementing devices.

Reply:
Hmm, that should not be happening (must say that the Todo stuff was added quite late in the release so it might have slipped, I'll double-check this when I'm at my desktop).

In this case I would just enable it. Add a device with entry type service and call it a day